storage/database/rocksdb_database.go

```diff
@@ -105,11 +105,26 @@ func NewRocksDB(path string, config *RocksDBConfig) (*rocksDB, error) {
 		return nil, err
 	}
 
+	// Ensure we have some minimal caching and file guarantees
+	if config.CacheSize < minCacheSizeForRocksDB {
+		config.CacheSize = minCacheSizeForRocksDB
```

Comment:
maybe print warn log?

Reply:
Added `Warn` log here and below as well.